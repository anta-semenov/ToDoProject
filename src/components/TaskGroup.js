import React from 'react'
import Task from './Task'

export default class TaskGroup extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (<div className='group'>
      {this.props.groupName ? <div className='group__name'>{this.props.groupName}</div> : null}
      <ul className='group__list'>
        {this.props.tasks.map(task =>
          <Task
            {...task}
            onClick = {() => this.props.onTaskClick(task.id)}
            onCheckboxClick = {() => this.props.onTaskCheckboxClick(task.id)}
          />
        )}
      </ul>
    </div>);
  }
}

TaskGroup.propTypes = {
  groupName: React.PropTypes.string,
  tasks: React.PropTypes.array.isRequired,
  onTaskClick: React.PropTypes.func.isRequired,
  onTaskCheckboxClick: React.PropTypes.func.isRequired
}
