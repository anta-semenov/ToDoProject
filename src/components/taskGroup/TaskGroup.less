@import '../../styles/variables.less';

.task-group {
  position: relative;
  margin: 0 0 30px 0;
}
.task-group__title {
  position: relative;
  font-size: 10px;
  font-weight: 400;
  text-transform: uppercase;
  color: @color-font-light;
}
.task-group__list {
  position: relative;
  margin: 0;
  padding: 0;
  list-style: none;
}
