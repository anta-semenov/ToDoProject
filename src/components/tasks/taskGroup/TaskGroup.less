@import '../../../styles/variables.less';
@import '../../../styles/commonStyles.less';

.task-group {
  position: relative;
  margin: 0 0 40px 0;
}
.task-group__title {
  .small__title();
  margin: 0 0 5px 5px;
  color: @color-font-medium;

  &.is-clickable {
    cursor: pointer;

    &:hover {
      color: @color-font-basic;
    }
  }
}
.task-group__list {
  position: relative;
  margin: 0;
  padding: 0;
  list-style: none;
}
