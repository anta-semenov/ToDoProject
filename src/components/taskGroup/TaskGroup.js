import React from 'react'
import Task from '../task/Task'
import PureRenderMixin from 'react-addons-pure-render-mixin'

import './TaskGroup.less'

export default class TaskGroup extends React.Component {
  constructor(props) {
    super(props)
    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this)
  }

  render() {
    return (
      <li className='task-group'>
        {this.props.groupTitle ? <div className='task-group__title'>{this.props.groupTitle}</div> : null}
        <ul className='task-group__list'>
          {this.props.tasks.map(task =>
            <Task
              key={`task-${task.get('id')}`}
              id={task.get('id')}
              title={task.get('title')}
              completed={task.get('completed')}
              today={task.get('today')}

              description={task.get('description')}
              priority={task.get('priority')}
              date={task.get('date')}
              active={this.props.activeItem === task.get('id')}
              latentToday={this.props.todayLatentTasks ? this.props.todayLatentTasks.includes(task.get('id')) : undefined}

              onTaskClick={this.props.onTaskClick}
              onTaskCheckboxClick={this.props.onTaskCheckboxClick}
              onTaskTodayClick={this.props.onTaskTodayClick}
              onPriorityClick={this.props.onPriorityClick}
            />
          )}
        </ul>
      </li>)
  }
}

TaskGroup.propTypes = {
  groupTitle: React.PropTypes.string,
  tasks: React.PropTypes.object.isRequired,
  activeItem: React.PropTypes.number,
  todayLatentTasks: React.PropTypes.object,
  onTaskClick: React.PropTypes.func.isRequired,
  onTaskCheckboxClick: React.PropTypes.func.isRequired,
  onTaskTodayClick: React.PropTypes.func.isRequired,
  onPriorityClick: React.PropTypes.func.isRequired
}
