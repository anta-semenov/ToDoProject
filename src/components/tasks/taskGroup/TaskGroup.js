import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { TransitionMotion, spring, presets } from 'react-motion'
import Task from '../task/Task'

import './TaskGroup.less'

const TaskGroup = ({ groupTitle, tasks, activeTask, latentTasks, trackingTask, ...rest}) => {
  const getDefaultStyles = () => tasks.toArray().map(task => ({
    key: task.get('id'),
    style: { opacity: 0, height: 0 },
    data: task
  }))
  const getStyles = () => tasks.toArray().map(task => ({
    key: task.get('id'),
    style: { opacity: spring(1), height: spring(46) },
    data: task
  }))
  const willEnter = () => ({ opacity: 0, height: 0 })
  const willLeave = () => ({ opacity: spring(0), height: spring(0) })

  return (
    <li className='task-group'>
      {groupTitle ? <div className='task-group__title'>{groupTitle}</div> : null}
      <TransitionMotion
        defaultStyles={getDefaultStyles()}
        styles={getStyles()}
        willEnter={willEnter}
        //willLeave={willLeave}
      >
        {styles => {
          return <ul className='task-group__list'>
            {styles.map(({ key, style, data }) => {
              return <Task
                {...rest}
                key={key}
                id={data.get('id')}
                style={style}

                active={activeTask === data.get('id')}
                latent={latentTasks ? latentTasks.has(data.get('id')) : undefined}
                tracking={trackingTask === data.get('id')}

                completed={data.get('completed')}
                today={data.get('today')}
                priority={data.get('priority')}
                someday={data.get('someday', false)}

                title={data.get('title')}
                description={data.get('description')}
                date={data.get('date') ? new Date(data.get('date')) : undefined}
              />
            })}
          </ul>
        }}
      </TransitionMotion>
    </li>
  )
}

TaskGroup.propTypes = {
  groupTitle: React.PropTypes.string,
  tasks: ImmutablePropTypes.listOf(
    ImmutablePropTypes.contains({
      id: React.PropTypes.string.isRequired,
      title: React.PropTypes.string.isRequired,
      completed: React.PropTypes.bool.isRequired,
      today: React.PropTypes.bool.isRequired,
      priority: React.PropTypes.string.isRequired,
      description: React.PropTypes.oneOfType([
        React.PropTypes.string,
        ImmutablePropTypes.map
      ]),
      date: React.PropTypes.number
    })
  ).isRequired,
  activeTask: React.PropTypes.string,
  latentTasks: ImmutablePropTypes.mapOf(React.PropTypes.number),
  trackingTask: React.PropTypes.string,

  onTaskClick: React.PropTypes.func.isRequired,
  onTaskCheckboxClick: React.PropTypes.func.isRequired,
  onTaskTodayClick: React.PropTypes.func.isRequired,
  onTaskPriorityClick: React.PropTypes.func.isRequired,
  onTaskTrackingClick: React.PropTypes.func.isRequired,
  onTaskSomedayClick: React.PropTypes.func.isRequired
}

export default TaskGroup
