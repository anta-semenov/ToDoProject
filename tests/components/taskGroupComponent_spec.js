import React from 'react'
import { shallow } from 'enzyme'
import { fromJS } from 'immutable'

import TaskGroup from '../../src/components/tasks/taskGroup/TaskGroup'

describe('TaskGroup component tests', () => {
  TaskGroup.__Rewire__(
    'Task',
    class extends React.Component {
      render() {
        return (
          <div className="task">
            <div className="task__title">{this.props.title}</div>
          </div>
        )
      }
    }
  )
  const props = {
    onTaskClick: () => {},
    onTaskCheckboxClick: () => {},
    onTaskTodayClick: () => {},
    onTaskPriorityClick: () => {},
    onTaskTrackingClick: () => {},
    addTaskToProject: () => {},
    addTaskContext: () => {}
  }

  test('Should render group name if it has one', () => {
    const taskGroup = shallow(<TaskGroup groupTitle="Test group" tasks={fromJS([])} {...props} />)
    expect(taskGroup).toMatchSnapshot()
  })
})
