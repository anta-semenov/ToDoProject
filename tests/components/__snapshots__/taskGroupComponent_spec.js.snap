// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskGroup component tests Should render group name if it has one 1`] = `
<li
  className="task-group"
>
  <div
    className="task-group__title"
  >
    Test group
  </div>
  <TransitionMotion
    defaultStyles={Array []}
    didLeave={[Function]}
    styles={Array []}
    willEnter={[Function]}
    willLeave={[Function]}
  />
</li>
`;
